python/pylibcudf/pylibcudf/hashing.pyx

```diff
@@ -0,0 +1,246 @@
+# Copyright (c) 2024, NVIDIA CORPORATION.
+from libc.stdint cimport uint32_t, uint64_t
+from libcpp.memory cimport unique_ptr
+from libcpp.utility cimport move
+from pylibcudf.libcudf.column.column cimport column
+from pylibcudf.libcudf.hash cimport (
+    DEFAULT_HASH_SEED,
+    md5 as cpp_md5,
+    murmurhash3_x64_128 as cpp_murmurhash3_x64_128,
+    murmurhash3_x86_32 as cpp_murmurhash3_x86_32,
+    sha1 as cpp_sha1,
+    sha224 as cpp_sha224,
+    sha256 as cpp_sha256,
+    sha384 as cpp_sha384,
+    sha512 as cpp_sha512,
+    xxhash_64 as cpp_xxhash_64,
+)
+from pylibcudf.libcudf.table.table cimport table
+
+from .column cimport Column
+from .table cimport Table
+
+LIBCUDF_DEFAULT_HASH_SEED = DEFAULT_HASH_SEED
+
+cpdef Column murmurhash3_x86_32(
+    Table input,
+    uint32_t seed=DEFAULT_HASH_SEED
+):
+    """Computes the MurmurHash3 32-bit hash value of each row in the given table.
+
+    For details, see :cpp:func:`murmurhash3_x86_32`.
+
+    Parameters
+    ----------
+    input : Table
+        The table of columns to hash
+    seed : uint32_t
+        Optional seed value to use for the hash function
+
+    Returns
+    -------
+    pylibcudf.Column
+        A column where each row is the hash of a row from the input
+    """
+    cdef unique_ptr[column] c_result
+    with nogil:
+        c_result = move(
+            cpp_murmurhash3_x86_32(
+                input.view(),
+                seed
+            )
+        )
+
+    return Column.from_libcudf(move(c_result))
+
+
+cpdef Table murmurhash3_x64_128(
+    Table input,
+    uint64_t seed=DEFAULT_HASH_SEED
+):
+    """Computes the MurmurHash3 64-bit hash value of each row in the given table.
+
+    For details, see :cpp:func:`murmurhash3_x64_128`.
+
+    Parameters
+    ----------
+    input : Table
+        The table of columns to hash
+    seed : uint32_t
+        Optional seed value to use for the hash function
+
+    Returns
+    -------
+    pylibcudf.Table
+        A table of two UINT64 columns
+    """
+    cdef unique_ptr[table] c_result
+    with nogil:
+        c_result = move(
+            cpp_murmurhash3_x64_128(
+                input.view(),
+                seed
+            )
+        )
+
+    return Table.from_libcudf(move(c_result))
+
+
+cpdef Column xxhash_64(
+    Table input,
+    uint64_t seed=DEFAULT_HASH_SEED
+):
+    """Computes the xxHash 64-bit hash value of each row in the given table.
+
+    For details, see :cpp:func:`xxhash_64`.
+
+    Parameters
+    ----------
+    input : Table
+        The table of columns to hash
+    seed : uint32_t
+        Optional seed value to use for the hash function
+
+    Returns
+    -------
+    pylibcudf.Column
+        A column where each row is the hash of a row from the input
+    """
+
+    cdef unique_ptr[column] c_result
+    with  nogil:
+        c_result = move(
```

Comment:
This `std::move` is unecessary. xref https://github.com/rapidsai/cudf/pull/16983